DQM/L1TMonitor/src/L1TStage2CaloLayer1.cc

```diff
@@ -348,92 +380,116 @@ void L1TStage2CaloLayer1::dqmAnalyze(const edm::Event& event,
       continue;
     }
 
-    // HCAL LLP trigger feature bits monitoring
+    // 6 HCAL fine grain bits from Layer1 readout
+    // Fg 0 is always there in ctp7 payload, set as bit 8 after Et bits in unpacked towerDatum
+    // When additionalFB flag is set:
+    // Fg 1-5 are unpacked and set as bits 0-5 in another unpacked towerDatum2 and packed in HCALTP sample(1)
+    // since standard sample size is 16bit and there is no room in sample(0) which contains already Et and link status
+    // Otherwise:
+    // Fg 1-5 are all zero
+    bool layer1fg0 = recdTp.SOI_fineGrain(0);
+    bool layer1fg1 = false;
+    bool layer1fg2 = false;
+    bool layer1fg3 = false;
+    bool layer1fg4 = false;
+    bool layer1fg5 = false;
+    if (additionalFB && (abs(ieta) < 29)) {
+      for (const auto& tp : (*hcalTPsRecd)) {
+        if (not(tp.id().ieta() == ieta && tp.id().iphi() == iphi)) {
+          continue;
+        }
+        layer1fg1 = tp.sample(1).raw() & (1 << 0);
+        layer1fg2 = tp.sample(1).raw() & (1 << 1);
+        layer1fg3 = tp.sample(1).raw() & (1 << 2);
+        layer1fg4 = tp.sample(1).raw() & (1 << 3);
+        layer1fg5 = tp.sample(1).raw() & (1 << 4);
+      }
+    }
+
+    // Check mismatches only for HBHE
+    const bool Hfg0Agreement = (abs(ieta) < 29) ? (layer1fg0 == uHTRfg0) : true;
+    const bool Hfg1Agreement = (abs(ieta) < 29) ? (layer1fg1 == uHTRfg1) : true;
+    const bool Hfg2Agreement = (abs(ieta) < 29) ? (layer1fg2 == uHTRfg2) : true;
+    const bool Hfg3Agreement = (abs(ieta) < 29) ? (layer1fg3 == uHTRfg3) : true;
+    const bool Hfg4Agreement = (abs(ieta) < 29) ? (layer1fg4 == uHTRfg4) : true;
+    const bool Hfg5Agreement = (abs(ieta) < 29) ? (layer1fg5 == uHTRfg5) : true;
+    // Mute fg4 and fg5 for now (reserved bits not used anyway)
+    const bool HfgAgreement = (Hfg0Agreement && Hfg1Agreement && Hfg2Agreement && Hfg3Agreement);
+
+    // Construct an 6-bit integer from the layer1 fine grain readout (input to 6:1 logic emulation)
+    uint64_t fg_bits = 0;
+    if (layer1fg0) {
+      fg_bits |= 0x1;
+    }
+    if (layer1fg1) {
+      fg_bits |= 0x1 << 1;
+    }
+    if (layer1fg2) {
+      fg_bits |= 0x1 << 2;
+    }
+    if (layer1fg3) {
+      fg_bits |= 0x1 << 3;
+    }
+    if (layer1fg4) {
+      fg_bits |= 0x1 << 4;
+    }
+    if (layer1fg5) {
+      fg_bits |= 0x1 << 5;
+    }
+
+    // Current 6:1 LUT in fw
+    const uint64_t HCalFbLUT = 0xBBBABBBABBBABBBA;
```

Comment:
I know @vshang recently opened a PR that included this LUT as a config param to the caloparams:
https://github.com/cms-sw/cmssw/pull/41046/files#diff-4664492025070575b58209e109e6c1fa97512eae8c07c841724000157c2a4703R171-R173
This is hard coded directly. Is this resilient enought to change it can be left constant and hard coded?

Reply:
hi @aloeliger, as I understood every time new LUT is uploaded, a new calo param config file will be created with new file name (like the new caloParams_2022_v0_6_cfi.py @vshang created in his PR) instead of overwriting some existing config file in production, so not sure if there is easy way to load LUT dynamically without knowing the next file name until created. But anyway we anticipate this BBBA LUT to stay for years unless HCAL decides to change their LLP algorithm in short time, so hard coded in DQM for now.